vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/endpoints/installer.go

```diff
@@ -139,15 +139,19 @@ func (a *APIInstaller) getResourceKind(path string, storage rest.Storage) (schem
 	}
 
 	object := storage.New()
-	fqKinds, _, err := a.group.Typer.ObjectKinds(object)
+	fqKinds, unversioned, err := a.group.Typer.ObjectKinds(object)
 	if err != nil {
 		return schema.GroupVersionKind{}, err
 	}
 
 	// a given go type can have multiple potential fully qualified kinds.  Find the one that corresponds with the group
 	// we're trying to register here
 	fqKindToRegister := schema.GroupVersionKind{}
+	unversionedCandidate := schema.GroupVersionKind{}
 	for _, fqKind := range fqKinds {
+		if unversioned && fqKind.Group == "" {
```

Comment:
I start to think that the `Status` type is actually wrong, compare https://github.com/openshift/origin/blob/34fd627194cf6b6e428641c3b0abceeb8938adf7/pkg/build/registry/buildconfig/webhook.go#L41. We return `Build` if I read that code correctly. This also matches the integration tests: https://github.com/openshift/origin/blob/8262531881906765c9982cfedc4396b9c8a74a6b/test/integration/webhook_test.go#L116-L116

@smarterclayton you wrote this ^^